### PR TITLE
Cleanup build env

### PR DESCRIPTION
**Pull request by neithernut**
This PR cleans up the build environment by grouping the objects/sources in modules.

Note that this PR conflicts with all PRs introducing files.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -138,8 +94,17 @@ endif()
...
+#
+# Pull in all the modules
+#
+foreach(MODULE IN LISTS WS_MODULES)
+    add_subdirectory(${MODULE})
+    set(SOURCE_FILES ${SOURCE_FILES} $<TARGET_OBJECTS:${MODULE}>)
+endforeach()
+
 #
-# Libreset will be a shared object
```
Comment (matthiasbeyer) [inline src/CMakeLists.txt:142]: LOOOL.

**Comment by matthiasbeyer** [comment]
Looks good to me, but I would be pleased if @TheNeikos could also ACK.

ACK.

**Comment by TheNeikos** [comment]
I suggested a small addition -> remove redundant file names in things like "wayland/wayland_buffer"

**Comment by matthiasbeyer** [comment]
Could be included, indeed!

**Comment by neithernut** [comment]
The renaming is done.

---

Note that the new modules may also be used in tests --if you need to pull in (nearly) all the sources of a module.

**Comment by TheNeikos** [comment]
ACK from me

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**